ext/standard/math.c

```diff
@@ -877,11 +882,41 @@ PHPAPI int _php_math_basetozval(zval *arg, int base, zval *ret)
 			c -= 'A' - 10;
 		else if (c >= 'a' && c <= 'z')
 			c -= 'a' - 10;
-		else
+		else if (c == '-')
+			negative = !negative;
+		else if (isspace(c)) {
+			if (num == 0 && fnum == 0) {
+				continue;
+			}
+
+			last_char = 1;
+			continue;
+		}
+		else {
+			invalid_chars++;
 			continue;
+		}
+
+		if (last_char == 1 && c > 0 && c < base) {
+			invalid_chars ++;
+		}
+
+		if (c >= base) {
+			if (i == length -1) {
+				if (base == 16 && c == 33) { /* allow the second char to be x */
```

Comment:
Only if the first char is `0`? I don't think we want to allow `9xAB` for hex strings.

Reply:
https://github.com/php/php-src/pull/4328  This has been addressed here